### ISSUE TITLE
Implement Aliases?

### ISSUE THREAD

**Issue by probonopd** [helloSystem/hello#89]
Unlike a symlink, the original file can change and an Alias will still work.

https://eclecticlight.co/2020/05/21/bookmarks-a-type-of-alias-their-access-and-use/

**Comment by grahamperrin** [comment]
There's support in **HFS Plus** – `kIsAlias`. 

To the best of my knowledge: nothing comparable in UFS or OpenZFS. 

### Reference

* [Technical Note TN1150: HFS Plus Volume Format](https://developer.apple.com/library/archive/technotes/tn/tn1150.html) in the Apple Developer Documentation Archive
* https://github.com/unsound/hfsexplorer/search?q=kIsAlias
* [fileXray User Guide and Reference](https://filexray.com/fileXray.pdf)
* [Finder Interface Reference](https://web.archive.org/web/20090817162236/http://developer.apple.com/documentation/Carbon/reference/Finder_Interface/Reference/reference.html) in the Wayback Machine (2009-08-17)
* https://superuser.com/a/222590/84988

**Comment by probonopd** [comment]
On the Mac, one can even make an Alias on FAT32. So it is not filesystem dependent. It is just a binary file. If we find a spec for it, we might be able to implement it.

This is how the file begins:

![image](https://user-images.githubusercontent.com/2480569/104088909-7a7dae00-526a-11eb-9eb1-0ed4cdd18127.png)

**Comment by grahamperrin** [comment]
Yeah, whilst in the kitchen the phrase _AliasManager_ came to mind. Whether you'll find relevant source code or spec, I have no idea.

**Comment by probonopd** [comment]
At least conceptually it is explained here: 
https://en.wikipedia.org/wiki/Alias_(Mac_OS)#File_structure

**Comment by grahamperrin** [comment]
Also via Wikipedia: [Alias Manager | Apple Developer Documentation](https://developer.apple.com/documentation/coreservices/carbon_core/alias_manager) (deprecated)

▶ https://developer.apple.com/library/archive/documentation/FileManagement/Conceptual/FileSystemProgrammingGuide/AccessingFilesandDirectories/AccessingFilesandDirectories.html#//apple_ref/doc/uid/TP40010672-CH3-SW10 (archived)

Via <https://github.com/nathanday/ndalias/issues/3#issuecomment-114761786>:

* https://developer.apple.com/library/mac/documentation/CoreFoundation/Reference/CFURLRef/index.html#//apple_ref/c/func/CFURLCreateBookmarkDataFromAliasRecord
